### PR TITLE
Sidebar component

### PR DESCRIPTION
**Pull request by kyleharper**
### 💬 Description
This PR adds components for `Sidebar`. A sidebar consists of;

- `header`
- `body`
- `section`(s)
- `heading` (s) (can optionally show a toggle which hooks into context for the section)
- `subheading` (s)
- `footer`

These create the following: https://share.getcloudapp.com/jkueGrqB

### ✅ Checklist
- [ ] Tests written
- [ ] Browser tested
- [ ] Added to documentation
- [ ] Added to storybook

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -106,7 +106,6 @@ export Pill from './Pill';
 export GuideCard from './GuideCard';
 export ApprovedLabel from './ApprovedLabel';
 export Person from './Person';
-export { Sidebar } from './Sidebar/Sidebar';
```
Comment (kyleharper) [inline lib/index.js:109]: This export is never used.

File: lib/src/components/sidebar/Sidebar.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { Layout } from 'lib';
+import { SidebarHeader } from './SidebarHeader';
+import { SidebarBody } from './SidebarBody';
+import { SidebarSection } from './SidebarSection';
+import { SidebarHeading } from './SidebarHeading';
+import { SidebarSubHeading } from './SidebarSubHeading';
+
+export function Sidebar({ children, className = '', ...rest }) {
+  return (
+    <Layout className={`sidebar ${className}`} {...rest}>
```
Comment (kyleharper) [inline lib/src/components/sidebar/Sidebar.js:11]: All `Sidebars` adopt the `Layout` to inherit the overflow scroll behaviour that we abstracted into `Layout.Body`.

The header & footer is then fixed as expected.

In our implementation we failed to do this (cause GUI didn't provide this out the box) and some sidebars don't overflow scroll 🤦

Comment (kyleharper) [reply lib/src/components/sidebar/Sidebar.js]: This is no longer the case.

File: lib/Layout/LayoutOverlaySidebar.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ import * as easings from 'd3-ease';
 import cx from 'classnames';
 import { useTransition, animated } from 'react-spring';
 
-export function Sidebar({
+export function LayoutOverlaySidebar({
```
Comment (kyleharper) [inline lib/Layout/LayoutOverlaySidebar.js:7]: The existing `Sidebar` is only ever used in `Layout` so I moved it next to that area to make it easier to understand and to separate the concerns of the contents of a sidebar and the layout aspect of it.

File: stories/components/Layout.js [hunk 4]
```diff
@@ -1,16 +1,21 @@
...
+import { Layout } from "../../lib";
+import { SidebarExample } from '../../lib/src/components/sidebar/SidebarStory';
 
 storiesOf('Components', module).add('Layout', () => {
   const isOpen = boolean('isOpen', false);
 
   return (
     <Layout>
+      <style>
+        {'#root { padding: 0 }'}
```
Comment (kyleharper) [inline stories/components/Layout.js:13]: Padding was throwing off the story as it's "full" screen.

File: styles/components/_ui.scss [hunk 5]
```diff
@@ -79,12 +79,14 @@
 
 @import '../../lib/ButtonNew/styles/styles.scss';
 
+@import '../../lib/src/prefabs/modalUpload/modalUpload';
+@import '../../lib/src/prefabs/files/fileCard/fileCard';
```
Comment (kyleharper) [inline styles/components/_ui.scss:83]: Moved prefabs with other prefabs.

**Review by DavidHollins6 — APPROVED** [review]

File: lib/src/components/sidebar/SidebarSection.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+
+export const SidebarSectionContext = React.createContext({});
+
+export function SidebarSection({ children, className = '' }, ...rest) {
```
Comment (AmeeMorris) [inline lib/src/components/sidebar/SidebarSection.js:5]: 🚩 `...rest` needs to be inside the curly braces

**Review by AmeeMorris — APPROVED** [review]
One comment on a typo, but the rest looks good to me!

**Pull request merged**